pkg/pillar/diskmetrics/diskmetrics.go

```diff
@@ -70,14 +70,15 @@ func CreateImg(ctx context.Context, log *base.LogObject, diskfile string, format
 	return nil
 }
 
-//ConvertImg do conversion of diskfile to outputFile with defined format
-func ConvertImg(ctx context.Context, log *base.LogObject, diskfile, outputFile, outputFormat string) error {
+//RolloutImgToBlock do conversion of diskfile to outputFile with defined format
+func RolloutImgToBlock(ctx context.Context, log *base.LogObject, diskfile, outputFile, outputFormat string) error {
 	if _, err := os.Stat(diskfile); err != nil {
 		return err
 	}
 	// writeback cache instead of default unsafe, out of order enabled, skip file creation
-	args := []string{"convert", "-t", "writeback", "-W", "-n", "-O", outputFormat, diskfile, outputFile}
-	output, err := base.Exec(log, "/usr/bin/qemu-img", args...).WithContext(ctx).CombinedOutput()
+	// Timeout 2 hours
+	args := []string{"convert", "--target-is-zero", "-t", "writeback", "-W", "-n", "-O", outputFormat, diskfile, outputFile}
+	output, err := base.Exec(log, "/usr/bin/qemu-img", args...).WithContext(ctx).CombinedOutputWithCustomTimeout(432000)
```

Comment:
sorry for commenting very late. Though this fix works I think we should have some intelligence build into it. We should adjust timeout parameter depending on size of the block device. Else if a smaller block size conversion is stuck for some reason we will wait for 2 hours or so.